### PR TITLE
Fix MessagingFilter NPE

### PR DESCRIPTION
**Pull request by EvanOxfeld**
- [x] @sfeng88 @rygim @jharwig 
- [x] @srfarley @joeferner
- [x] @EvanOxfeld @joeybrk372
- [x] @mwizeman @dsingley

Fixes null pointer exception preventing the session expiration message from reaching the client.

v5analytics/visallo#53

### CODE REVIEW

**Comment by EvanOxfeld** [comment]
Should `MessagingFilter` be unit tested?

**Comment by srfarley** [comment]
Yes, please unit test if possible.

File: web/web-base/src/main/java/org/visallo/web/MessagingFilter.java [hunk 1]
```diff
@@ -36,48 +37,102 @@ public BroadcastAction filter(String broadcasterId, AtmosphereResource r, Object
...
+        if (type != null) {
+            switch (type) {
+                case TYPE_SET_ACTIVE_WORKSPACE:
+                    return false;
+                case TYPE_SESSION_EXPIRATION:
+                    return true;
+            }
+        }
+
+        if (session == null) {
```
Comment (srfarley) [inline web/web-base/src/main/java/org/visallo/web/MessagingFilter.java]: `return session != null && !shouldRejectMessageByPermissions(json, session);`

Comment (EvanOxfeld) [inline web/web-base/src/main/java/org/visallo/web/MessagingFilter.java]: @srfarley Done. Changed to `return session != null && shouldSendMessageByPermissions(json, session);`

Comment (joeferner) [inline web/web-base/src/main/java/org/visallo/web/MessagingFilter.java]: I don't particularly like this change. These two tests are for different reasons. Combining them just makes it harder to read in my opinion.

Comment (EvanOxfeld) [inline web/web-base/src/main/java/org/visallo/web/MessagingFilter.java]: @srfarley @joeferner: How about a compromise?

```
if (session == null) {
  return false;
}

return shouldSendMessageByPermissions(json, session);
```

Comment (srfarley) [inline web/web-base/src/main/java/org/visallo/web/MessagingFilter.java]: It's a matter of style. I usually like to combine conditions with two terms, and don't find it harder to read.

Comment (srfarley) [inline web/web-base/src/main/java/org/visallo/web/MessagingFilter.java]: It doesn't matter to me. I'm not going to hold up this PR over this.

Comment (joeferner) [inline web/web-base/src/main/java/org/visallo/web/MessagingFilter.java]: No need to hold up the PR I just think these two statements have different purposes and just as I would split out two classes to handle responsibilities I would split out these statements as well if on the other hand these two statements were working together then I would join them it feels like a run on sentence to me.

Comment (EvanOxfeld) [inline web/web-base/src/main/java/org/visallo/web/MessagingFilter.java]: Merging with the compromise code snippet above. That said, I defer to you all on Java code style if you would like to change anything.

**Pull request merged**